index.js

```diff
@@ -166,7 +166,8 @@ Router.prototype.handle = function handle(req, res, callback) {
   // manage inter-router variables
   var parentParams = req.params
   var parentUrl = req.baseUrl || ''
-  var done = restore(callback, req, 'baseUrl', 'next', 'params')
+  var parentMatchedRoutes = req.matchedRoutes
+  var done = restore(callback, req, 'baseUrl', 'next', 'params', 'matchedRoutes')
```

Comment:
It seems like adding `matchedRoutes` is necessary here, but no tests fail. So we should either try to add a test for it, or remove it if it's not necessary.

Reply:
It should be apparent once more tests are added with sub routers. This controls the behavior of what happens on a router exit.